samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs

```diff
@@ -109,64 +152,165 @@ private MessagingExtensionActionResponse ShareMessageCommand(ITurnContext<IInvok
+                                        Card = GetAdaptiveCardAttachmentFromFile("justintimeinstallation.json"),
+                                        Height = 200,
+                                        Width = 400,
+                                        Title = "Adaptive Card - App Installation",
+                                    },
+                                },
+                            };
+                        }
+                        throw; // It's a different error.
+                    }
+
                     return new MessagingExtensionActionResponse
                     {
                         Task = new TaskModuleContinueResponse
                         {
                             Value = new TaskModuleTaskInfo
                             {
-                                Card = GetAdaptiveCardAttachmentFromFile("justintimeinstallation.json"),
+                                Card = GetAdaptiveCardAttachmentFromFile("adaptiveCard.json"),
                                 Height = 200,
                                 Width = 400,
-                                Title = "Adaptive Card - App Installation",
+                                Title = $"Welcome {memberName}",
                             },
                         },
                     };
-                }
-                throw; // It's a different error.
             }
+        }
 
-            return new MessagingExtensionActionResponse
+        private MessagingExtensionActionResponse DateDayInfo(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action)
+        {
+            var response = new MessagingExtensionActionResponse()
+            {
+                Task = new TaskModuleContinueResponse()
+                {
+                    Value = new TaskModuleTaskInfo()
+                    {
+                        Height = 175,
+                        Width = 300,
+                        Title = "Task Module Razor View",
+                        Url = baseUrl + "/Home/RazorView",
+                    },
+                },
+            };
+            return response;
+        }
+
+        private MessagingExtensionActionResponse TaskModuleHTMLPage(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action)
+        {
+            var response = new MessagingExtensionActionResponse()
             {
-                Task = new TaskModuleContinueResponse
+                Task = new TaskModuleContinueResponse()
                 {
-                    Value = new TaskModuleTaskInfo
+                    Value = new TaskModuleTaskInfo()
                     {
-                        Card = GetAdaptiveCardAttachmentFromFile("adaptiveCard.json"),
                         Height = 200,
                         Width = 400,
-                        Title = $"Welcome {memberName}",
+                        Title = "Task Module HTML",
+                        Url = baseUrl + "/htmlTaskModule.html",
```

Comment:
either the page needs to be named 'htmlTaskModule.html' or this line should read `Url = baseUrl + "/htmlpage",`
![image](https://user-images.githubusercontent.com/11055362/121607448-e7a59300-ca04-11eb-96d0-2fff9d1108d9.png)